pkg/chunkenc/interface.go

```diff
@@ -1,22 +1,23 @@
 package chunkenc
 
 import (
-	"errors"
 	"io"
+	"net/http"
 	"time"
 
 	"github.com/grafana/loki/pkg/iter"
 	"github.com/grafana/loki/pkg/logproto"
 	"github.com/grafana/loki/pkg/logql"
+	"github.com/grafana/loki/pkg/util"
 )
 
 // Errors returned by the chunk interface.
 var (
-	ErrChunkFull       = errors.New("chunk full")
-	ErrOutOfOrder      = errors.New("entry out of order")
-	ErrInvalidSize     = errors.New("invalid size")
-	ErrInvalidFlag     = errors.New("invalid flag")
-	ErrInvalidChecksum = errors.New("invalid checksum")
+	ErrChunkFull       = util.NewCodedError(http.StatusInternalServerError, "chunk full")
```

Comment:
I feel like it's more when you use a method of this package you should check:

```
if err := chunk.Append(e);err {
     if err == chunk. ErrOutOfOrder {
          sampler.Add(500,err)
     }else{
        sampler.Add(400,err)
     }
}
```

So I don't think this change should be made here.